src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -156,13 +173,45 @@ assign(SyntheticEvent.prototype, {
    * `PooledClass` looks for `destructor` on each instance it releases.
    */
   destructor: function() {
+    var setNull = setToNullOrWarning.bind(this);
     var Interface = this.constructor.Interface;
-    for (var propName in Interface) {
-      this[propName] = null;
+    for (var interfacePropName in Interface) {
+      setNull(interfacePropName);
+    }
+    otherNullableProps.forEach(setNull);
+    this._destructored = true;
+
+    function setToNullOrWarning(propName) {
```

Comment:
Unless I'm missing something obvious, putting a function declaration in an if block like that wont work. However dead code elimination would remove it anyway, so I think we're good.